### ISSUE TITLE
[CM11 nightly 20140320] mempodroid vulnerability

### ISSUE THREAD

**Issue by 9omorra** [LegacyXperia/local_manifests#535]
According to X-Ray from http://www.xray.io/, the latest LX11 is vulnerable to mempodroid on Xperia Pro (MK16i, iyokan). The installed firmware is cm-11.0-20140320-NIGHTLY-LegacyXperia-iyokan.zip.

The relevant kernel patch to fix this issue is probably http://git.kernel.org/cgit/linux/kernel/git/torvalds/linux.git/commit/?id=e268337dfe26dfc7efd422a804dbb27977a3cccc
The issue depends on the kernel version, but it probably does not only affect iyokan.

(In contrast to that, the old FXP250-cm-10-20131215-UNOFFICIAL-iyokan.zip was not vulnerable to mempodroid, as it uses a kernel <2.6.39).

**Comment by mikeNG** [comment]
I'll check that

**Comment by mikeNG** [comment]
The above commit is already present in our kernel since it's merged in 3.2 and we use 3.4

**Comment by 9omorra** [comment]
So either the patch does not work, or the x-ray test is buggy. Yet it seems that X-Ray is buggy:
X-Ray 1.0 seems to be from 2012/07/22. And there's an article about X-Ray from Jon Oberheide on Sep 12, 2012, i.e. more recent than latest X-Ray. Comments from https://www.duosecurity.com/blog/early-results-from-x-ray-over-50-of-android-devices-are-vulnerable say:

```
Tom Tom • a year ago
 Updated - the kernel code has the fix for /proc/<pid>/mem. I'm not sure why X-Ray 
 still reports vulnerability</pid>

    Jon Oberheide Mod Tom • a year ago
    Yeah, it's a known false positive, going to be fixed in the next release (also was 
    filtered out from our above results calculations). Some folks fixed the problem 
    like Linux mainline did while AOSP did the proper thing and actually just removed 
    mem_write completely.

        Tom Jon Oberheide • a year ago
         Thank you very much for confirm it.
```

So, afaics the X-Ray mempodroid test is buggy in current X-Ray V1.0. If you agree, feel free to close this bugreport and sorry for the noise.

**Comment by mikeNG** [comment]
Yes, this issue is fixed since 2012/android 4.0.4
